src/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp

```diff
@@ -40,61 +40,54 @@ using namespace Aws::Utils::Logging;
 #ifndef WINHTTP_OPTION_HTTP_PROTOCOL_USED
     static const DWORD WINHTTP_OPTION_HTTP_PROTOCOL_USED = 134;
 #endif
+#ifndef WINHTTP_PROTOCOL_FLAG_HTTP2
+    static const DWORD WINHTTP_PROTOCOL_FLAG_HTTP2 = 0x1;
+#endif
+#ifndef WINHTTP_PROTOCOL_FLAG_HTTP3
+    static const DWORD WINHTTP_PROTOCOL_FLAG_HTTP3 = 0x2;
+#endif
 
-DWORD ConvertHttpVersionToWinHttpVersion(const Aws::Http::Version version)
+std::initializer_list<DWORD> GetWinHttpVersionsToTry(const Aws::Http::Version version)
```

Comment:
I don't think that initializer_list works in such context (return by value from a method), it is probably an undefined behavior here.
I'd suggest returning a const reference to a local static const object, such as
```
const std::initializer_list<DWORD>& GetWinHttpVersionsToTry(const Aws::Http::Version version) {
...
    else if (version == Version::HTTP_VERSION_3)
    {
      static const std::initializer_list<DWORD> HTTP123{ WINHTTP_PROTOCOL_FLAG_HTTP3 | WINHTTP_PROTOCOL_FLAG_HTTP2, WINHTTP_PROTOCOL_FLAG_HTTP2};
      return HTTP123;
    }
```
or use some other approach.

Please check the cpp reference:

> Copying a std::initializer_list does not copy the [backing array](https://en.cppreference.com/w/cpp/language/list_initialization#List-initializing_std::initializer_list) of the corresponding initializer list.

https://en.cppreference.com/w/cpp/utility/initializer_list

see also: https://stackoverflow.com/questions/15286450/lifetime-of-a-stdinitializer-list-return-value

Reply:
Done.